app-modules/report/src/Filament/Widgets/MostRecentStudentsTable.php

```diff
@@ -78,21 +78,25 @@ public function table(Table $table): Table
 
                 $startDate = $this->getStartDate();
                 $endDate = $this->getEndDate();
-
-                return Student::whereIn($key, function ($query) use ($key, $startDate, $endDate) {
-                    $query->select($key)
-                        ->from((new Student())->getTable())
-                        ->whereNotNull('created_at_source')
-                        ->whereNull('deleted_at')
-                        ->when(
-                            $startDate && $endDate,
-                            function (Builder $query) use ($startDate, $endDate): Builder {
-                                return $query->whereBetween('created_at_source', [$startDate, $endDate]);
-                            }
-                        )
-                        ->orderBy('created_at_source', 'desc')
-                        ->take(100);
-                })
+                $segmentId = $this->getSelectedSegment();
+
+                $subQuery = Student::query()
+                    ->select($key)
+                    ->whereNotNull('created_at_source')
+                    ->whereNull('deleted_at')
+                    ->when(
+                        $startDate && $endDate,
+                        fn (Builder $query) => $query->whereBetween('created_at_source', [$startDate, $endDate])
+                    )
+                    ->when(
+                        $segmentId,
+                        fn (Builder $query) => $this->segmentFilter($query, $segmentId)
+                    )
+                    ->orderBy('created_at_source', 'desc')
+                    ->take(100);
+
+                return Student::query()
+                    ->whereIn($key, $subQuery)
```

Comment:
No need to change this here, I just wanted to point out that a `whereIn` DOES make sense here because we are specifically controlling this down to `100` records. So that makes sure we don't overload any of this.